### PR TITLE
Bump hdf5 to v1.12

### PR DESCRIPTION
**Pull request by epassaro**
### :pencil: Description

**Type:** :beetle: `bugfix` |  :roller_coaster: `infrastructure`

Trying to fix the annoying bug in the documentation pipeline. The `hdf5` v1.10 package from conda-forge hasn't been updated in more than two years, so I'm proposing update it to v1.12 (latest is 1.14). 

### :pushpin: Resources

See: https://anaconda.org/conda-forge/hdf5/files

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [x] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: tardis_env3.yml [hunk 1]
```diff
@@ -8,17 +8,17 @@ dependencies:
   # WARNING: any change to this section must be applied to the conda-forge
   # package recipe at https://github.com/conda-forge/tardis-sn-feedstock
 
-  - python=3.8
+  - python =3.8
   - setuptools
-  - setuptools_scm<7
+  - setuptools_scm <7
```
Comment (andrewfullard) [inline tardis_env3.yml:13]: If you're changing the formatting, why not spaces on both sides of the operator?

**Comment by andrewfullard** [comment]
Why not the latest hdf5? Does that involve changing other dependencies?

File: tardis_env3.yml (hunk 1 above)
Comment (epassaro) [reply tardis_env3.yml:13]: It's the conda-forge standard, I thought would be a good idea to keep the same format.

**Comment by epassaro** [comment]
> Why not the latest hdf5? Does that involve changing other dependencies?

I was being conservative, v1.14 was launched a few months ago. I'm open to changing it if everyone agrees.

**Review by andrewfullard — APPROVED** [review]

**Review by atharva-2001 — APPROVED** [review]

**Pull request merged**